CMakeLists.txt

```diff
@@ -339,6 +339,10 @@ endif()
 
 if(APPLE)
   set(DEFAULT_OMP_FLAG "-Xclang -fopenmp")
+  execute_process(COMMAND brew --prefix
```

Comment:
This won't allow for anyone to build OpenMP themselves...
So maybe only do all of this if `OpenMP_ROOT` is undefined?

Also, I'm not much of a Mac person, so I'm not sure if it makes a difference - in the CI we do the following to identify the full `OpenMP_ROOT` - is `prefix/opt/libomp` always going to be the correct path?
https://github.com/AdaptiveCpp/AdaptiveCpp/blob/0359cac9e08d44c48c586ef3ec88e95ac64a4d44/.github/workflows/macos.yml#L23

Reply:
Hi there, thanks for the review, I haven't looked into the yml yesterday but I've modified the `execute_process`  cmake command to have the same effect as `brew list libomp | grep libomp.a | sed -E "s/\/lib\/.*//"`.

Also add an IF(NOT DEFINED OpenMP_ROOT) to make sure other people can build it

Reply:
Hi, thanks!
I just realized that still will override manually specified hints such as the environment variable `OMP_ROOT`.
So, maybe it makes sense to move this logic to where we're actually looking for OpenMP and first doing an `find_package(OpenMP QUIET)`, and if that fails, add the `OpenMP_ROOT` and search with `REQUIRED`.

The relevant location will be: https://github.com/AdaptiveCpp/AdaptiveCpp/blob/0359cac9e08d44c48c586ef3ec88e95ac64a4d44/src/runtime/CMakeLists.txt#L280

To show this working in CI, maybe remove the line I referenced earlier from the Action yml... The CI should the still work with your changes.

Reply:
hmm i think on MacOS, CMake can only find OpenMP if either OMP_ROOT or OpenMP_ROOT is defined. When I did what you suggested first with find_package(OpenMP QUIET) before searching for OpenMP_ROOT, CMake can't seem to find the OpenMP package.

I also removed the old checks in CMakelists and remove the OMP_ROOT on yml